codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -83,9 +85,14 @@
         void onWeightChanged(Http2Stream stream, short oldWeight);
 
         /**
-         * Called when a GO_AWAY frame has either been sent or received for the connection.
+         * Called when a GO_AWAY frame was sent for the connection.
+         */
+        void goAwaySent(long errorCode, ByteBuf debugData);
+
+        /**
+         * Called when a GO_AWAY frame was received for the connection.
          */
-        void goingAway();
+        void goAwayReceived(long errorCode, ByteBuf debugData);
```

Comment:
@nmittler - Sry for the delay.  I just want to be careful as we introduce duplication in our interfaces.  I'm not disputing its potential convenience as [2] described.  Having the "sent" interface makes sense because this interface is unique and the event can be automatically generated by the codec.  I don't see the logical difference between this an `onGoAwayRead` but I am fine with keeping both interfaces if you feel strongly about it.  We should consider documenting the relationship/behavior between the two interfaces. Keeping them decoupled may make sense but when someone does discover that they both exist I think questions will arise...better to address these explicitly so it is known to be intentional and the rational/behavior is well defined.

> We could potentially remove the "redundancy" by just not calling that onGoAwayRead method from the decoder (and document that of course).

I think keeping the `onGoAwayRead` makes sense.  It very closely maps to frame types from the specification and should be very familiar to anyone familiar with the spec.